sources/osgViewer/View.js

```diff
@@ -245,12 +245,19 @@ View.prototype = {
     },
 
     setSceneData: function(node) {
-        if (node === this._scene.getSceneData()) return;
+        var previousNode = this._scene.getSceneData();
+        if (node === previousNode) return;
 
         this._scene.setSceneData(node);
 
+        var children = this._camera.getChildren();
+        var statsNode = undefined;
+        for (var i = 0, l = children.length; i < l; i++) {
+            if (children[i].getName() === 'osgStats') statsNode = children[i];
```

Comment:
You can break the loop after finding it. But maybe better to add a helper node in Camera to get the statsNode?